pages/agent/v3/hooks.md.erb

```diff
@@ -101,12 +101,16 @@ a plugin can define.
 
 ### Creating agent lifecycle hooks
 
-Agent lifecycle hooks can be written in the programming language of your choice
-and are executed by the Buildkite agent. See the documentation for each agent
-lifecycle hook for details on the interface between them and the Buildkite
-agent.
-
+The Buildkite agent executes agent lifecycle hooks.
```

Comment:
How about 

> The Buildkite agent can run agent lifecycle hooks before bootstrapping each agent, and after shutting down each agent. 

Although it runs them if they are there, not every if they aren't

Reply:
And it looks like my understanding might not be correct here. Or the description might be out.

Reply:
> Executed before any job is started. Useful for [adding strict checks](https://bk-docs-pr-1628.onrender.com/docs/agent/v3/securing#strict-checks-using-a-pre-bootstrap-hook) before jobs are permitted to run. 

I **think** this means run when the agent starts, before it runs any jobs. It doesn't mean runs before each job. Right?

Reply:
OK, that makes sense